lib/utils.js

```diff
@@ -0,0 +1,46 @@
+var Q = require('q');
+var es = require('.././config/es');
+
+var log = console.log;
+
+exports = module.exports = {
+  waitForElastic: function (waitCnt) {
+    var waitCntDef = 50;
+    waitCnt = waitCnt || 50;  // 40 requests by default
+
+    var deferred = Q.defer();
+
+    (function pinger(done) {
+      if(waitCnt != waitCntDef) log('requests left', waitCnt);
+
+      if(waitCnt-- == 0) {
+        deferred.reject(new Error('Max requests exceeded, elastic is dead.'));
```

Comment:
Aha. Tried and completely remembered the exact purpose of `setTimeout`.
`requestTimeout` is really for requests and seems it doesn't work in case of elastic not running yet (which is the case for vagga).
Look at this:
> $ vagga app
Entered 1444925667160
Wait counter is 1999
...
Wait counter is 3
Wait counter is 2
Wait counter is 1
Wait counter is 0
Error: { '0': 'Max requests exceeded 1444925668156' }